src/matlab2tikz.m

```diff
@@ -4235,11 +4208,39 @@ function warnFacetedInterp(m2t)
             ['{' arrowHeadSize '/1000*\pgfplotspointmetatransformed}']);
 
         headStyle = ['-{Straight Barb[' opts_print(arrowHeadOptions) ']}'];
-        quiverOptions = opts_add(quiverOptions, 'every arrow/.append style', ...
-                                 ['{' headStyle '}']);
+        plotOptions = opts_add(plotOptions, headStyle);
+        signalDependency(m2t, 'tikzlibrary', 'arrows.meta');
+    else
+        plotOptions = opts_add(plotOptions, '-');
+    end
+
+    % Append the arrow style to the TikZ options themselves.
+    color = get(h, 'Color');
+    [m2t, lineOptions] = getLineOptions(m2t, h);
+    [m2t, arrowcolor] = getColor(m2t, h, color, 'patch');
+    plotOptions = opts_add(plotOptions, 'color', arrowcolor);
+    plotOptions = opts_merge(plotOptions, lineOptions);
+
+    % Define the quiver settings
+    quiverOptions = opts_new();
+    quiverOptions = opts_add(quiverOptions, 'u', '\thisrow{u}');
+    quiverOptions = opts_add(quiverOptions, 'v', '\thisrow{v}');
+    if is3D
+        quiverOptions = opts_add(quiverOptions, 'w', '\thisrow{w}');
+        arrowLength = '{sqrt((\thisrow{u})^2+(\thisrow{v})^2+(\thisrow{w})^2)}';
+    else
+        arrowLength = '{sqrt((\thisrow{u})^2+(\thisrow{v})^2)}';
     end
+    plotOptions = opts_add(plotOptions, 'point meta', arrowLength);
+    plotOptions = opts_add(plotOptions, 'point meta min', '0');
+
     plotOptions = opts_addSubOpts(plotOptions, 'quiver', quiverOptions);
 
+    % Check whether there is a legend and add a legend image option without scaling
+    if ~isempty(m2t.legendHandles)
```

Comment:
Is there a reason why you check for `m2t.legendHandles` and do not use the function `hasLegendEntry` instead?